src/components/navbar/search_page.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import SearchSubreddits from './searching';
+import SavedSubreddits from './show_saved_subreddits';
+import ButtonComponent from "./visibility_button";
+import {Link} from "react-router";
+const CSSTransitionGroup = require('react-transition-group/CSSTransitionGroup');
+
+interface IProps {
+  dispatch?: any;
+  isVisible?: boolean;
+  quanityOfSavedSubreddits?: number;
+  whereYouAre?: string;
+}
+
+const mapStateToProps = (state: any): IProps => {
+  return {
+    isVisible: state.subreddits.isVisible,
+    quanityOfSavedSubreddits: state.subreddits.savedSubreddits.length,
+    whereYouAre: state.routing.locationBeforeTransitions.pathname,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any) => {
+  return {dispatch};
+};
+
+@(connect(mapStateToProps, mapDispatchToProps) as any)
+export default class Navbar extends React.Component<IProps, {}> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+    };
+
+    document.addEventListener('touchmove', this.preventMacbookBack);
+  }
+
+  private preventMacbookBack = (e: any) => {
+    e.preventDefault();
+  };
+
+  public printLocation = () => {
+    const locationName = this.props.whereYouAre.split('/');
+    if (locationName.length <= 2)
+    {
+      return 'Reddit Front Page';
+    }
+    if (locationName.length >= 3)
+    {
+      locationName.splice(3, locationName.length);
+    }
+    return (
+      <Link to={locationName.join('/') + '/'}>
+        {locationName.join('/')}
+      </Link>
+    );
+  };
+
+  public render() {
+    const authLink = 'https://www.reddit.com/api/v1/authorize?client_id=YJa94gbugND2ZQ&response_type=code&state=42asdf&redirect_uri=http://localhost:8000/auth&duration=permanent&scope=identity';
```

Comment:
Should be moved into config and set as `AUTH_LINK` const imported from constants.